static/js/search.js

```diff
@@ -0,0 +1,186 @@
+    
+                //Get the contents of the entire document
+                var text = document.getElementById("doc-view").value;
+                // Split text into tokens
+                var token = text.split(/\s/);
+    
+                //Current position in the document
+                var position = 0;
+      
+                token.forEach( function(word){
+                    if(word.match(regex)){
+                        console.log("Found " + word + "\nStart postion: " + position + "\nEnd Position: " + (position + word.length));
+                        let range = {
+                            'startPosition': position,
+                            'endPosition': position + word.length
+                        };
+                        let annotationCreated = tagModel.addAnnotation(range, tagModel.currentCategory);
+                        console.log("Highlighted: " + range.startPosition + "-" + range.endPosition);
+                        $('#recent').text(annotationCreated.content.trunc(50, true));
+                        addToList(annotationCreated.content);
+                    }
+                    position += word.length+1; // the plus 1 accounts for the space at the end
+                })
+                renderTextareaHighlights(); 
+                }    
+        } else if ($("#search-type").children("option:selected").val() === "delete") {
+            console.log("Searching to delete.");
+            var searching = document.getElementById("search-box").value;
+            if (tagModel.currentCategory === null) {
+                alert('Error: Please select a category first');
+                return;
+            } else if (searching == "" || searching.trim() == ""){
+                alert("Please enter the text you want to highlight");
+                return;
+            } else {
+                let regex = null;
+                if($("#text-type").children("option:selected").val() === "plain-text"){
+                    regex = new RegExp("\\b" + searching + "\\b");
+                } else {
+                    regex = new RegExp("\\b" + RegExp.escape(searching) + "\\b", flags.join(''));
+                }
+                console.log("Searching for " + searching + " in document"); 
+    
+                //Get the annotations of the current document
+                
+                tagModel.currentDoc.annotations.forEach( function(annotation){
+                    if(annotation.content.match(regex)){
+                        console.log("Found " + annotation.content + "\nStart postion: " + position + "\nEnd Position: " + (position + annotation.content.length));
+                        tagModel.removeAnnotation(annotation);
+                    }
+                    position += annotation.content.length+1; // the plus 1 accounts for the space at the end
+                })
+                renderTextareaHighlights();
+            }
+        }
```

Comment:
Duplicate code.
Move to a function and call the function instead.

Reply:
stopped having a stroke and fixed it in the latest commit